### DISCUSSION TITLE
Is a puck still required if using Home Assistant?

### DISCUSSION THREAD

**Discussion by brystmar** [RobertD502/home-assistant-flair#74] [Q&A] [answered]
I'm debating an investment into these Flair vents.  I already have temp sensors in each room and know precisely how I want to control these using automations in Home Assistant.  Will the vents work by themselves via HA, or do I still need a puck?

**Comment by RobertD502** [comment, marked as answer]
The vents themselves don't have the means for communicating with Flair's servers. In order to use Flair vents, you will need AT LEAST 1 puck. This puck will need to be powered by being plugged into an outlet to act as the "gateway puck" (pucks can also be powered via batteries, but when they are powered by batteries they can't act as a gateway). Flair pucks have both WiFi capabilities (used to connect to your network and send data about the puck and vents paired with it to Flair's servers) and a radio transceiver used by the puck to communicate with Flair vents. Flair vents only have the radio required to communicate with Flair pucks.

Essentially:

Flair Vent(s) <----RF communication ----> Flair Puck (Gateway) <----Comm. over WAN ---> Flair Servers

The Flair puck, which receives data from the Flair vents, communicates with Flair's servers to let them know the state of the vents.
If you use the Flair app or this HA integration (which uses the cloud-based Flair API) to, for example, "Open bedroom vents", this request is sent to Flair's servers, who then tell the puck to command the bedroom vents to open, and the puck sends the command, via RF, to the bedroom vents.

I believe, but don't quote me on this, a single puck can be paired with up to 20 vents. However, this will really depend on how far away the vents you want to pair with the puck are.

So, at minimum, you need 1 Puck to use multiple Flair vents (NOT 1 puck per vent).

TLDR: At least 1 puck is required if you want to use Flair vents.

The benefit of using the Home Assistant integration is that you don't have to use only temperature sensors officially supported by Flair or buy additional pucks (in addition to your gateway puck) to act as temperature sensors in order to control Flair vents since you can use any temperature sensors in your Home Assistant automations to determine how you want to control your Flair vents.

**Comment by brystmar** [comment]
Appreciate the technical overview!
Makes sense from an architecture perspective. Was hoping to save a few
bucks (and an extra piece of hardware). Looks like there’s a puck in my
future.
